### PR TITLE
YOLO Family: Data Loaders 

### PR DESCRIPTION
**Pull request by The-Indian-Chinna**
This pull request adds the detection data loader, box utilities, iou utilities, preprocessing operations and input test to the YOLOv3 and v4 families.

## Type of change

For a new feature or function, please create an issue first to discuss it
with us before submitting a pull request.

Note: Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation update
- [ ] TensorFlow 2 migration
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] A new research paper code implementation
- [ ] Other (Specify)

## Tests

**Test Configuration**:

Tests are provided for the detection pipeline that utilizes all of the files in the pull request. The test provided preprocesses the coco dataset and allows the user to see the outputted preprocessed image with bounding boxes drawn on to the image. This test is not automated and the  user will have to click through a set of images and check the validity of the images manually.

## Checklist

- [x] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [x] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [x] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [x] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by The-Indian-Chinna** [comment]
@googlebot I consent.

**Comment by vishnubanna** [comment]
@googlebot I consent.

**Comment by anivegesana** [comment]
@googlebot I consent.

**Comment by Tyan3001** [comment]
I consent.

**Comment by anivegesana** [comment]
@googlebot I consent.

File: official/vision/beta/projects/yolo/utils/iou_ops.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
+"""Intersection over union calculation utils."""
+
+# Import libraries
```
Comment (arashwan) [inline official/vision/beta/projects/yolo/utils/iou_ops.py:3]: Remove comment

File: official/vision/beta/projects/yolo/utils/iou_ops.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import tensorflow as tf
+import math
+
```
Comment (arashwan) [inline official/vision/beta/projects/yolo/utils/iou_ops.py:6]: can you merge iou_ops and box_ops and add tests.

File: official/vision/beta/projects/yolo/utils/iou_ops.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        iou, diou = compute_diou(box1, box2)
+
+        # computer aspect ratio consistency
+        v = box_utils.aspect_ratio_consistancy(box1[..., 2], box1[..., 3], box2[..., 2],
+                                      box2[..., 3])
+
+        # compute IOU regularization
+        a = tf.math.divide_no_nan(v, ((1 - iou) + v))
+        ciou = diou + v * a
+    return iou, ciou
```
Comment (arashwan) [inline official/vision/beta/projects/yolo/utils/iou_ops.py:116]: new line.

File: official/vision/beta/projects/yolo/utils/preprocessing_ops.py [hunk 4]
```diff
@@ -0,0 +1,177 @@
+import tensorflow as tf
+import tensorflow.keras.backend as K
```
Comment (arashwan) [inline official/vision/beta/projects/yolo/utils/preprocessing_ops.py:2]: You don't have to use tensorflow.keras.backend. Can you just use tf.keras ?

File: official/vision/beta/projects/yolo/utils/preprocessing_ops.py [hunk 5]
```diff
@@ -0,0 +1,177 @@
+import tensorflow as tf
```
Comment (arashwan) [inline official/vision/beta/projects/yolo/utils/preprocessing_ops.py:1]: add tests.

File: official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py [hunk 6]
```diff
@@ -0,0 +1,209 @@
+""" Detection Data parser and processing for YOLO.
```
[the remark this answers is not in the document]
Comment (The-Indian-Chinna) [reply official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py:1]: Addressed

File: official/vision/beta/projects/yolo/utils/preprocessing_ops.py (hunk 4 above)
Comment (The-Indian-Chinna) [reply official/vision/beta/projects/yolo/utils/preprocessing_ops.py:2]: Addressed

File: official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+          bounding box for data augmentation during training.
+        net_down_scale: an `int` that down scales the image width and height
+          to the closest multiple of net_down_scale.
+        max_process_size: an `int` for maximum image width and height.
+        min_process_size: an `int` for minimum image width and height ,
+        max_num_instances: an `int` number of maximum number of instances in
+          an image.
+        random_flip: a `bool` if True, augment training with random
+          horizontal flip.
+        pct_rand: an `int` that prevents do_scale from becoming larger than
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py]: Fixed in fcaf5fb604a22fe08553bb573b6901efcafa4ad6

File: official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+    Args:
+        data: a dict of Tensors produced by the decoder.
+    Returns:
+        images: the image tensor.
+        labels: a dict of Tensors that contains labels.
+    """
+
+    shape = tf.shape(data["image"])
+    image = data["image"] / 255
+    image = tf.image.resize(image,
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/9321d835e596f5a2aee481c6584d1f74b40f8327

File: official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py]: Fixed in f28269c8be3c616251af66317f53bf4a4f75f187

File: official/vision/beta/projects/yolo/dataloaders/classification_tfds_decoder.py [hunk 9]
```diff
@@ -26,11 +25,22 @@ def __init__(self):
     return
 
   def decode(self, serialized_example):
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/dataloaders/classification_tfds_decoder.py]: Fixed f28269c8be3c616251af66317f53bf4a4f75f187

File: official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+                            (self._max_process_size, self._max_process_size))
+    image = tf.image.random_brightness(image=image, max_delta=.1)  # Brightness
+    image = tf.image.random_saturation(image=image, lower=0.75,
+                                       upper=1.25)  # Saturation
+    image = tf.image.random_hue(image=image, max_delta=.1)  # Hue
+    image = tf.clip_by_value(image, 0.0, 1.0)
+    boxes = data["groundtruth_boxes"]
+    image_shape = tf.shape(image)[:2]
+
+    self._random_flip = False
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py]: Fixed in https://github.com/tensorflow/models/commit/70f736a16d3d8cbf71e99e415b0c66083c342a72

File: official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+    Returns:
+        images: the image tensor.
+        labels: a dict of Tensors that contains labels.
+    """
+
+    shape = tf.shape(data["image"])
+    image = data["image"] / 255
+    image = tf.image.resize(image,
+                            (self._max_process_size, self._max_process_size))
+    image = tf.image.random_brightness(image=image, max_delta=.1)  # Brightness
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py]: Fixed in 9321d835e596f5a2aee481c6584d1f74b40f8327

File: official/vision/beta/projects/yolo/dataloaders/dataloaders_test.py [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+import tensorflow_datasets as tfds
+from matplotlib import patches
+from matplotlib import pyplot as plt
+from official.vision.beta.projects.yolo.dataloaders.decoders.tfds_coco_decoder import \
+    MSCOCODecoder
+from official.vision.beta.projects.yolo.utils.box_utils import xcycwh_to_xyxy
+
+from .YOLO_Detection_Input import Parser
+
+matplotlib.use("TkAgg")
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/dataloaders/dataloaders_test.py]: Fixed in https://github.com/tensorflow/models/commit/9321d835e596f5a2aee481c6584d1f74b40f8327

File: official/vision/beta/projects/yolo/dataloaders/dataloaders_test.py [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+    wh = bbox[2:4] - bbox[0:2]
+    rect = patches.Rectangle(bbox[0:2] * 416,
+                             wh[0] * 416,
+                             wh[1] * 416,
+                             linewidth=1,
+                             edgecolor="r",
+                             facecolor="none")
+    ax.add_patch(rect)
+    # Add the patch to the Axes
+  plt.show()
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/dataloaders/dataloaders_test.py]: Fixed in https://github.com/tensorflow/models/commit/9321d835e596f5a2aee481c6584d1f74b40f8327

File: official/vision/beta/projects/yolo/dataloaders/dataloaders_test.py [hunk 14]
```diff
@@ -0,0 +1,37 @@
+import matplotlib
+import tensorflow_datasets as tfds
+from matplotlib import patches
+from matplotlib import pyplot as plt
+from official.vision.beta.projects.yolo.dataloaders.decoders.tfds_coco_decoder import \
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/dataloaders/dataloaders_test.py]: Fixed in https://github.com/tensorflow/models/commit/ddb99d95566037d20361e21c4d4641622e434494

File: official/vision/beta/projects/yolo/dataloaders/dataloaders_test.py [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+    MSCOCODecoder
+from official.vision.beta.projects.yolo.utils.box_utils import xcycwh_to_xyxy
+
+from .YOLO_Detection_Input import Parser
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/dataloaders/dataloaders_test.py]: Fixed in https://github.com/tensorflow/models/commit/ddb99d95566037d20361e21c4d4641622e434494

File: official/vision/beta/projects/yolo/dataloaders/decoders/image_net_decoder.py [hunk 16]
```diff
@@ -0,0 +1,21 @@
+from official.vision.beta.dataloaders import decoder
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/dataloaders/decoders/image_net_decoder.py]: Fixed in https://github.com/tensorflow/models/commit/66c4f91562ac5b198c98fbff966959eb683db18b

File: official/vision/beta/projects/yolo/utils/box_ops.py [hunk 17]
```diff
@@ -0,0 +1,172 @@
+""" bounding box utils file """
+
+# import libraries
+import tensorflow as tf
+import math
+from typing import *
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/utils/box_ops.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/f28269c8be3c616251af66317f53bf4a4f75f187

File: official/vision/beta/projects/yolo/utils/box_ops.py [hunk 18]
```diff
@@ -0,0 +1,172 @@
...
+
+
+def yxyx_to_xcycwh(box: tf.Tensor):
+    """Converts boxes from ymin, xmin, ymax, xmax to x_center, y_center, width, height.
+    Args:
+        box: a `Tensor` whose last dimension is 4 representing the coordinates of boxes
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/utils/box_ops.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/f28269c8be3c616251af66317f53bf4a4f75f187

File: official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py [hunk 19]
```diff
@@ -0,0 +1,209 @@
...
+Parse image and ground truths in a dataset to training targets and package them
+into (image, labels) tuple for RetinaNet.
+"""
+
+# Import libraries
+import tensorflow as tf
+from official.vision.beta.dataloaders import parser
+from official.vision.beta.ops import box_ops, preprocess_ops
+from official.vision.beta.projects.yolo.utils import preprocessing_ops
+from official.vision.beta.projects.yolo.utils import box_ops as box_utils
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py]: Fixed in 2306e11d3c44ee6f54b92f52cbf4ab7dab7102ba

File: official/vision/beta/projects/yolo/utils/box_ops.py [hunk 20]
```diff
@@ -0,0 +1,172 @@
...
+        width = xmax - xmin
+        height = ymax - ymin
+        box = tf.concat([x_center, y_center, width, height], axis=-1)
+    return box
+
+
+def xcycwh_to_yxyx(box: tf.Tensor, split_min_max: bool = False):
+    """Converts boxes from x_center, y_center, width, height to ymin, xmin, ymax, xmax.
+    Args:
+        box: a `Tensor` whose last dimension is 4 representing the coordinates of boxes in
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/utils/box_ops.py]: Fixed in 2306e11d3c44ee6f54b92f52cbf4ab7dab7102ba

File: official/vision/beta/projects/yolo/utils/iou_ops.py (hunk 2 above)
Comment (anivegesana) [reply official/vision/beta/projects/yolo/utils/iou_ops.py]: Fixed in https://github.com/tensorflow/models/commit/f28269c8be3c616251af66317f53bf4a4f75f187

File: official/vision/beta/projects/yolo/utils/iou_ops.py (hunk 1 above)
Comment (anivegesana) [reply official/vision/beta/projects/yolo/utils/iou_ops.py]: Fixed in https://github.com/tensorflow/models/commit/f28269c8be3c616251af66317f53bf4a4f75f187

File: official/vision/beta/projects/yolo/utils/iou_ops.py (hunk 3 above)
Comment (anivegesana) [reply official/vision/beta/projects/yolo/utils/iou_ops.py]: Fixed in https://github.com/tensorflow/models/commit/f28269c8be3c616251af66317f53bf4a4f75f187

File: official/vision/beta/projects/yolo/dataloaders/dataloaders_test.py [hunk 21]
```diff
@@ -0,0 +1,37 @@
...
+
+dataset = tfds.load("coco", split="train")
+
+decoder = MSCOCODecoder()
+parser = Parser()
+
+dataset = dataset.map(decoder.decode)
+imgs = dataset.map(parser._parse_train_data)
+
+for img, label in imgs.take(10):
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/dataloaders/dataloaders_test.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/36136cdd6dfc07594b91bda0dd8e5189e81d3e75

File: official/vision/beta/projects/yolo/dataloaders/yolo_input_test.py [hunk 22]
```diff
@@ -0,0 +1,88 @@
...
+  global_batch_size: int = 10
+  is_training: bool = True
+  dtype: str = "float16"
+  decoder = None
+  parser: Parser = Parser()
+  shuffle_buffer_size: int = 10000
+  tfds_download: bool = True
+
+
+def test_yolo_input():
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/dataloaders/yolo_input_test.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/36136cdd6dfc07594b91bda0dd8e5189e81d3e75

File: official/vision/beta/projects/yolo/utils/preprocessing_ops.py (hunk 5 above)
Comment (anivegesana) [reply official/vision/beta/projects/yolo/utils/preprocessing_ops.py]: Fixed in 5d40ec4ec5fe258206aa887e06e2f90082c22d46

File: official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py [hunk 23]
```diff
@@ -0,0 +1,209 @@
...
+
+    randscale = self._image_w // self._net_down_scale
+
+    if not self._fixed_size:
+      do_scale = tf.greater(
+          tf.random.uniform([], minval=0, maxval=1, seed=self._seed),
+          1 - self._pct_rand)
+      if do_scale:
+        randscale = tf.random.uniform([],
+                                      minval=10,
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/a22cdffb9a9b2726f1ae252dd151a21cdef27da6

File: official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py [hunk 24]
```diff
@@ -0,0 +1,209 @@
...
+      images: the image tensor.
+      labels: a dict of Tensors that contains labels.
+    """
+
+    shape = tf.shape(data["image"])
+    image = tf.convert_to_tensor(data["image"])
+    image = tf.image.resize(image, size=(self._image_w, self._image_h))
+    image = image / 255
+    boxes = box_utils.yxyx_to_xcycwh(data["groundtruth_boxes"])
+    best_anchors = preprocessing_ops.get_best_anchor(boxes,
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/a22cdffb9a9b2726f1ae252dd151a21cdef27da6

File: official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py [hunk 25]
```diff
@@ -0,0 +1,209 @@
...
+                                      maxval=21,
+                                      seed=self._seed,
+                                      dtype=tf.int32)
+
+    if self._jitter_boxes != 0.0:
+      boxes = box_ops.denormalize_boxes(boxes, image_shape)
+      boxes = box_ops.jitter_boxes(boxes, 0.025)
+      boxes = box_ops.normalize_boxes(boxes, image_shape)
+
+    boxes = box_utils.yxyx_to_xcycwh(boxes)
```
[the remark this answers is not in the document]
Comment (anivegesana) [reply official/vision/beta/projects/yolo/dataloaders/YOLO_Detection_Input.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/a22cdffb9a9b2726f1ae252dd151a21cdef27da6

File: official/vision/beta/projects/yolo/ops/box_ops.py [hunk 26]
```diff
@@ -0,0 +1,275 @@
+""" bounding box utils file """
+
+# import libraries
```
Comment (arashwan) [inline official/vision/beta/projects/yolo/ops/box_ops.py:3]: remove the line.

File: official/vision/beta/projects/yolo/ops/preprocessing_ops.py [hunk 27]
```diff
@@ -0,0 +1,434 @@
...
+  pad_tensor = tf.fill(nshape, tf.cast(pad_value, dtype=value.dtype))
+  value = tf.concat([value, pad_tensor], axis=pad_axis)
+  return value
+
+
+def fit_preserve_aspect_ratio(image,
+                              boxes,
+                              width=None,
+                              height=None,
+                              target_dim=None):
```
Comment (arashwan) [inline official/vision/beta/projects/yolo/ops/preprocessing_ops.py:114]: add docstring

File: official/vision/beta/projects/yolo/ops/preprocessing_ops.py [hunk 28]
```diff
@@ -0,0 +1,434 @@
...
+  boxes = tf.concat([x, y, w, h], axis=-1)
+
+  boxes = box_ops.xcycwh_to_yxyx(boxes)
+  image = tf.image.resize(image, (target_dim, target_dim))
+  return image, boxes
+
+
+def get_best_anchor(y_true, anchors, width=1, height=1):
+  """
+    get the correct anchor that is assoiciated with each box using IOU betwenn
```
Comment (arashwan) [inline official/vision/beta/projects/yolo/ops/preprocessing_ops.py:154]: typo: between.

File: official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py [hunk 29]
```diff
@@ -0,0 +1,309 @@
...
+      if not batch:
+        mask[key] = preprocessing_ops.build_grided_gt(
+            raw_true, self._masks[key], width // 2**int(key), self._num_classes,
+            raw_true['bbox'].dtype, use_tie_breaker)
+      else:
+        mask[key] = preprocessing_ops.build_batch_grided_gt(
+            raw_true, self._masks[key], width // 2**int(key), self._num_classes,
+            raw_true['bbox'].dtype, use_tie_breaker)
+
+      mask[key] = tf.cast(mask[key], self._dtype)
```
Comment (arashwan) [inline official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py:119]: You don't want to cast labels to dtype, only the image.

Comment (The-Indian-Chinna) [reply official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/28559cab79d1ed7521a803844e096c5e79064bef

File: official/vision/beta/projects/yolo/ops/preprocessing_ops.py (hunk 28 above)
Comment (The-Indian-Chinna) [reply official/vision/beta/projects/yolo/ops/preprocessing_ops.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/28559cab79d1ed7521a803844e096c5e79064bef

File: official/vision/beta/projects/yolo/ops/box_ops.py (hunk 26 above)
Comment (The-Indian-Chinna) [reply official/vision/beta/projects/yolo/ops/box_ops.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/28559cab79d1ed7521a803844e096c5e79064bef

File: official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py [hunk 30]
```diff
@@ -0,0 +1,309 @@
...
+
+    if not self._fixed_size:
+      do_scale = tf.greater(
+          tf.random.uniform([], minval=0, maxval=1, seed=self._seed), 0.5)
+      if do_scale:
+        # This scales the image to a random multiple of net_down_scale
+        # between 320 to 608
+        randscale = tf.random.uniform([],
+                                      minval=320 // self._net_down_scale,
+                                      maxval=608 // self._net_down_scale,
```
[the remark this answers is not in the document]
Comment (The-Indian-Chinna) [reply official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/28559cab79d1ed7521a803844e096c5e79064bef

File: official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py [hunk 31]
```diff
@@ -0,0 +1,309 @@
...
+    randscale = self._image_w // self._net_down_scale
+
+    if not self._fixed_size:
+      do_scale = tf.greater(
+          tf.random.uniform([], minval=0, maxval=1, seed=self._seed), 0.5)
+      if do_scale:
+        # This scales the image to a random multiple of net_down_scale
+        # between 320 to 608
+        randscale = tf.random.uniform([],
+                                      minval=320 // self._net_down_scale,
```
[the remark this answers is not in the document]
Comment (The-Indian-Chinna) [reply official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/28559cab79d1ed7521a803844e096c5e79064bef

File: official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py [hunk 32]
```diff
@@ -0,0 +1,309 @@
...
+               anchors=None,
+               seed=10,
+               dtype='float16'):
+    """Initializes parameters for parsing annotations in the dataset.
+    Args:
+      image_w: a `Tensor` or `int` for width of input image.
+      image_h: a `Tensor` or `int` for height of input image.
+      num_classes: a `Tensor` or `int` for the number of classes.
+      fixed_size: a `bool` if True all output images have the same size.
+      jitter_im: a `float` that is the maximum jitter applied to the image for
```
[the remark this answers is not in the document]
Comment (The-Indian-Chinna) [reply official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/28559cab79d1ed7521a803844e096c5e79064bef

File: official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py [hunk 33]
```diff
@@ -0,0 +1,309 @@
...
+    self._random_flip = random_flip
+
+    self._aug_rand_saturation = aug_rand_saturation
+    self._aug_rand_brightness = aug_rand_brightness
+    self._aug_rand_zoom = aug_rand_zoom
+    self._aug_rand_hue = aug_rand_hue
+
+    self._seed = seed
+
+    if dtype == 'float16':
```
[the remark this answers is not in the document]
Comment (The-Indian-Chinna) [reply official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/28559cab79d1ed7521a803844e096c5e79064bef

File: official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py [hunk 34]
```diff
@@ -0,0 +1,309 @@
...
+        data augmentation during training.
+      jitter_boxes: a `float` that is the maximum jitter applied to the bounding
+        box for data augmentation during training.
+      net_down_scale: an `int` that down scales the image width and height to
+        the closest multiple of net_down_scale.
+      max_process_size: an `int` for maximum image width and height.
+      min_process_size: an `int` for minimum image width and height ,
+      max_num_instances: an `int` number of maximum number of instances in an image.
+      random_flip: a `bool` if True, augment training with random horizontal flip.
+      masks: a `Tensor`, `List` or `numpy.ndarrray` for anchor masks.
```
[the remark this answers is not in the document]
Comment (The-Indian-Chinna) [reply official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/28559cab79d1ed7521a803844e096c5e79064bef

File: official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py [hunk 35]
```diff
@@ -0,0 +1,309 @@
...
+               min_process_size=320,
+               max_num_instances=200,
+               random_flip=True,
+               aug_rand_saturation=True,
+               aug_rand_brightness=True,
+               aug_rand_zoom=True,
+               aug_rand_hue=True,
+               anchors=None,
+               seed=10,
+               dtype='float16'):
```
[the remark this answers is not in the document]
Comment (The-Indian-Chinna) [reply official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/28559cab79d1ed7521a803844e096c5e79064bef

File: official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py [hunk 36]
```diff
@@ -0,0 +1,309 @@
...
+from official.vision.beta.projects.yolo.ops import box_ops as yolo_box_ops
+
+
+class Parser(parser.Parser):
+  """Parser to parse an image and its annotations into a dictionary of tensors."""
+
+  def __init__(self,
+               image_w=416,
+               image_h=416,
+               num_classes=80,
```
[the remark this answers is not in the document]
Comment (The-Indian-Chinna) [reply official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/28559cab79d1ed7521a803844e096c5e79064bef

File: official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py [hunk 37]
```diff
@@ -0,0 +1,309 @@
...
+from official.vision.beta.ops import box_ops, preprocess_ops
+from official.vision.beta.projects.yolo.ops import preprocessing_ops
+from official.vision.beta.projects.yolo.ops import box_ops as yolo_box_ops
+
+
+class Parser(parser.Parser):
+  """Parser to parse an image and its annotations into a dictionary of tensors."""
+
+  def __init__(self,
+               image_w=416,
```
[the remark this answers is not in the document]
Comment (The-Indian-Chinna) [reply official/vision/beta/projects/yolo/dataloaders/yolo_detection_input.py]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/28559cab79d1ed7521a803844e096c5e79064bef

File: official/vision/beta/projects/yolo/ops/preprocessing_ops.py (hunk 27 above)
Comment (The-Indian-Chinna) [reply official/vision/beta/projects/yolo/ops/preprocessing_ops.py:121]: Fixed in https://github.com/tensorflow/models/pull/9493/commits/bb11fe05a6a24b084766659e149431d0f4fc1da4

**Review by arashwan — APPROVED** [review]

**Pull request merged**